src/core/PlayArea.ttslua

```diff
@@ -64,84 +49,35 @@ function log(message)
-function getClueCount(object, isFaceDown, playerCount)
-  local details = getLocation(object)
-  if details == nil then
-    error('attempted to get clue for unexpected object: ' .. object.getName())
-  end
-
-  log(object.getName() .. ' : ' .. details['type'] .. ' : ' .. details['value'] .. ' : ' .. details['clueSide'])
-  if ((isFaceDown and details['clueSide'] == 'back') or (not isFaceDown and details['clueSide'] == 'front')) then
-    if details['type'] == 'fixed' then
-      return details['value']
-    elseif details['type'] == 'perPlayer' then
-      return details['value'] * playerCount
-    end
-    error('unexpected location type: ' .. details['type'])
-  end
-  return 0
-end
-
-function spawnCluesAtLocation(clueCount, object)
-  if spawnedLocationGUIDs[object.getGUID()] ~= nil then
-    error('tried to spawn clue for already spawned location:' .. object.getName())
-  end
-
-  log('spawning clues for ' .. object.getName() .. '_' .. object.getGUID())
-  log('player count is ' .. getInvestigatorCount() .. ', clue count is ' .. clueCount)
-
-  -- mark this location as spawned, can't happen again
-  spawnedLocationGUIDs[object.getGUID()] = true
-
-  -- spawn clues (starting top right, moving to the next row after 4 clues)
-  local pos = object.getPosition()
-  for i = 1, clueCount do
-    local row = math.floor(1 + (i - 1) / 4)
-    local column = (i - 1) % 4
-    spawnClue({ pos.x + 1.5 - 0.55 * row, pos.y, pos.z - 0.825 + 0.55 * column })
-  end
-end
-
-function spawnClue(position)
-  local token = spawnObject({
-    position = position,
-    rotation = { 3.88, -90, 0.24 },
-    type     = 'Custom_Tile'
-  })
-
-  token.setCustomObject(clueData)
-  token.scale { 0.25, 1, 0.25 }
-  token.use_snap_points = false
-end
-
+-- Called by Custom Data Helpers to push their location data into the Data Helper.  This adds the
+-- data to the local token manager instance.
+---@param args Table Single-value array holding the GUID of the Custom Data Helper making the call
 function updateLocations(args)
-  custom_data_helper_guid = args[1]
-  local custom_data_helper = getObjectFromGUID(args[1])
-
-  for k, v in pairs(custom_data_helper.getTable("LOCATIONS_DATA")) do
-    LOCATIONS[k] = v
+  local customDataHelper = getObjectFromGUID(args[1])
+  if customDataHelper ~= nil then
+    tokenManager.addLocationData(customDataHelper.getTable("LOCATIONS_DATA"))
   end
 end
 
 function onCollisionEnter(collision_info)
   if not COLLISION_ENABLED then return end
 
   -- check if we should spawn clues here and do so according to playercount
-  local object = collision_info.collision_object
-  if getLocation(object) ~= nil and spawnedLocationGUIDs[object.getGUID()] == nil then
-    local clueCount = getClueCount(object, object.is_face_down, getInvestigatorCount())
-    if clueCount > 0 then spawnCluesAtLocation(clueCount, object) end
+  local card = collision_info.collision_object
+  if shouldSpawnTokens(card) then
+    tokenManager.spawnForCard(card)
```

Comment:
It could, but it would add some complexity that I don't think is necessary.  The JSON parser may be slow, but our metadata is still small and doing it for a specific card shouldn't add any appreciable time.  Even the worst case for this (placing a scenario that drops as many as 20) will be overwhelmed by the spawn time in TTS.

We should definitely keep an eye on this and see if it starts causing problems, but until it does I consider it a premature optimization.